nexus/reconfigurator/preparation/src/lib.rs

```diff
@@ -80,8 +80,11 @@ impl PlanningInputFromDb<'_> {
         datastore: &DataStore,
     ) -> Result<PlanningInput, Error> {
         opctx.check_complex_operations_allowed()?;
+        // Note we list *all* rows here including the ones for decommissioned
+        // sleds, because parts of the system consult the input to determine
+        // whether a sled is decommissioned.
         let sled_rows = datastore
-            .sled_list_all_batched(opctx, SledFilter::Commissioned)
+            .sled_list_all_batched(opctx, SledFilter::All)
```

Comment:
Just to confirm:

It seems like we *could* infer that a sled is decommissioned by seeing "a trace of it exists in the blueprint, but it is absent from the planning input".

Is this decision to make things explicit a safety-guard? Just trying to check my understanding whether this is "truly a necessity" or "an explicit alternative to relying on implied state".

Reply:
Yes, it's meant to be a safety guard specifically for cleanup -- because zone cleanup will actually remove entries from the blueprint. So it's possible that if a sled is missed somehow then it disappears from the blueprint (I actually first realized this while writing tests for Nexus zone cleanup).